src/Microsoft.Health.Dicom.Core/Features/CustomTag/ICustomTagStore.cs

```diff
@@ -0,0 +1,21 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.Dicom.Core.Features.CustomTag
+{
+    public interface ICustomTagStore
+    {
+        /// <summary>
+        /// Refresh given CustomTagList.
+        /// </summary>
+        /// <param name="customTags">The customTags.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>True if the custom tags has been refreshed, false otherwise</returns>
+        Task<bool> TryRefreshCustomTags(out CustomTagList customTags, CancellationToken cancellationToken = default);
```

Comment:
Why do we need the API to Try? Why not just

CustomTagList GetCustomTagsAsync()?

Reply:
the API need to handle 2 case:
1. If given customTags is latest, won't update the customTags, and return false which means not Refreshed
2. if given customTags is not latest, will update the customTags, and return true which means the customTags is refreshed.

CustomTagList GetCustomTagsAsync() doesn't have a good way to differentiate these 2.

Reply:
Store layer is just a dum data access layer, basically an interface that can be easily replaced when datastore changes. Rest of the logic should be implemented in different interfaces in the core